openhtf/io/output/mfg_inspector.py

```diff
@@ -430,18 +434,31 @@ def UploadTestRun(testrun, destination, credentials=None):
         raise UploadFailedError(resp, content)
 
   def __call__(self, test_record):  # pylint: disable=invalid-name
-    credentials = oauth2client.client.SignedJwtAssertionCredentials(
-        service_account_name=self.user,
-        private_key=self.keydata,
-        scope=self.SCOPE_CODE_URI,
-        user_agent='OpenHTF Guzzle Upload Client',
-        token_uri=self.token_uri)
-    credentials.set_store(self._MemStorage())
-
     testrun = _TestRunFromTestRecord(test_record)
+
     try:
+      credentials = oauth2client.client.SignedJwtAssertionCredentials(
+          service_account_name=self.user,
+          private_key=self.keydata,
+          scope=self.SCOPE_CODE_URI,
+          user_agent='OpenHTF Guzzle Upload Client',
+          token_uri=self.token_uri)
+      credentials.set_store(self._MemStorage())
       self.UploadTestRun(testrun, self.DESTINATION_URL, credentials)
-    except UploadFailedError:
+    except:
       # For now, just log the exception.  Once output is a bit more robust,
       # we can propagate this up and handle it accordingly.
+      if self._error_message:
+        sys.stdout.write(''.join([self._error_message, '\n']))
```

Comment:
If it's just one variable going into the string I would consider
```sys.stdout.write('%s\n' % self._error_message)```
instead of the empty string join here.